### PR TITLE
[test_hash] Enable backend topologies

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Let `test_hash` run on storage backend topo.
The failure reason is that because there is no default route on the DUT in the storage backend testbed, the IO packet that aims to test hashing functionality will not be correctly forwarded.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

#### How did you do it?
1. add a fixture `add_default_route_to_dut` to add the default route to the DUT.
2. modify ptf test `hash_test.py` to select `base_mac` randomly instead of using the `(0, 0)`(which doesn't exist on the backend topologies)

### CODE REVIEW

**Comment by lolyu** [comment]
```
```

**Comment by lolyu** [comment]
depends on https://github.com/Azure/sonic-mgmt/pull/3727

**Comment by lolyu** [comment]
```
fib/test_fib.py::test_hash[ipv4] PASSED                                                                                                                                                                                                                                                     [ 50%]
fib/test_fib.py::test_hash[ipv6] PASSED                                                                                                                                                                                                                                                     [100%]

=================================================================================================================================== 2 passed in 3271.55 seconds ===================================================================================================================================
```

**Review by wangxin — APPROVED** [review]

File: tests/fib/test_fib.py [hunk 1]
```diff
@@ -499,7 +502,48 @@ def ipver(request):
...
+    """
+    if "backend" in tbinfo["topo"]["name"]:
+        logging.info("Add default route on the DUT.")
+        try:
+            for duthost in duthosts:
+                cfg_facts = config_facts[duthost.hostname]
+                for asic_index, asic_cfg_facts in enumerate(cfg_facts):
+                    asic = duthost.asic_instance(asic_index)
+                    bgp_neighbors = asic_cfg_facts["BGP_NEIGHBOR"]
+                    ipv4_cmd_parts = ["ip route add default"]
```
Comment (neethajohn) [inline tests/fib/test_fib.py:519]: Are we adding default route for backend asics as well in multi-asic? Backend asics are skipped in the remove path in L538

Comment (lolyu) [reply tests/fib/test_fib.py:519]: Yes, I followed the logic here:
https://github.com/Azure/sonic-mgmt/blob/6af8a5c871a2ed4791ef9a72b3903a1b26cc08d8/tests/fib/test_fib.py#L42-L51

**Pull request merged**